### PR TITLE
pybricks.hubs.PrimeHub: Added a cleanup example.

### PR DESCRIPTION
**Pull request by lobodpav**

### CODE REVIEW

**Comment by lobodpav** [comment]
@dlech Does this one make sense?

**Comment by dlech** [comment]
I will defer to @laurensvalk on this one. This example isn't really specific to any one hub, so there might be a better place for it.

**Comment by laurensvalk** [comment]
> I will defer to @laurensvalk on this one. This example isn't really specific to any one hub, so there might be a better place for it.

I've commented something along these lines yesterday. Is it not visible?

**Comment by dlech** [comment]
No, it doesn't appear to be. Maybe need to click the "Finish your review" button under the "Files changed" tab?

**Review by laurensvalk — COMMENTED** [review]
Re-submitting review in case it was lost between notifications.

File: examples/pup/hub_primehub/cleanup.py [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+
+
+# Wrap the code in try-finally to ensure the cleanup code is always executed
+# no matter if the program gets stopped gracefully by the user or by an error
+try:
+    main()
+finally:
+    # The cleanup code goes here
+    hub.display.off()
+    hub.light.off()
```
[the remark this answers is not in the document]
Comment (lobodpav) [reply examples/pup/hub_primehub/cleanup.py]: This was my original intention, although I did not do so for some reason 🤦 

Anyway, the example now turns off ultrasonic and colour sensors' lights.

**Comment by lobodpav** [comment]
> Re-submitting review in case it was lost between notifications.

Your previous review did not show up for some reason.

File: examples/pup/hub_primehub/cleanup.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (lobodpav) [reply examples/pup/hub_primehub/cleanup.py]: @laurensvalk Does it look good now so we can merge the PR?

**Comment by laurensvalk** [comment]
Thanks for submitting this! It's been merged. To make sure it can be used on all hubs I have:

- Moved the example to the ultrasonic sensor.
- Used `Port.A`
- Used a print command instead of the prime hub display.
- Turned comments into full sentences consistent with other examples.
- Edited to comply with `flake8`.

**Comment by lobodpav** [comment]
> Thanks for submitting this! It's been merged. To make sure it can be used on all hubs I have:
> 
> * Moved the example to the ultrasonic sensor.
> * Used `Port.A`
> * Used a print command instead of the prime hub display.
> 
> And:
> 
> * Turned comments into full sentences consistent with other examples.
> * Edited to comply with `flake8`.

Well, the major goal of this PR was to eliminate the dazzling colour sensor 😄 Which is gone now 😢

**Comment by laurensvalk** [comment]
Turning off the matrix is done automatically, and having something that works on all hubs seems a bit nicer.

While examples of generic Python techniques like `try/finally` can be really useful, they are probably out of scope for this documentation. So an example with a specific use case like turning the lights off fits a bit better. 

> Well, the major goal of this PR was to eliminate the dazzling colour sensor.

The [original issue](https://github.com/pybricks/support/issues/574) referenced the ultrasonic sensor and this PR has no description, so I went with that. I'm happy to move the example to the color sensor though. Updated now.

**Pull request merged**